### DISCUSSION TITLE
Useable without npm?

### DISCUSSION THREAD

**Discussion by MakotoE** [typescript-language-server/typescript-language-server#703] [Q&A] [answered]
I see that you can specify `npmLocation` for the Automatic Type Acquisition feature. Without npm installed, you get the message `[tsclient] 'npm' is not recognized as an internal or external command, operable program or batch file.`. If you use typescript-language-server without npm installed, will it still run fine? Or will it disable certain protocol features?

**Comment by rchl** [comment, marked as answer]
I've fairly sure that Npm should only be needed for the automatic type acquisition. I say "fairly" because that's functionality that's embedded into the `typescript` itself and I'm not bothered to check its code base.

Automatic type acquisition is not an essential feature and it's fine to run without it (I personally do that myself). Just make sure that your project explicitly includes all needed `@types/*`.

You can even disable the additional process dedicated for it by passing a `disableAutomaticTypingAcquisition: true` option to the server. That should get rid of the warning also.
